src/PatientChecking/PatientChecking/Feature/Appointment/Queries/GetAppointmentPagingQuery.cs

```diff
@@ -0,0 +1,56 @@
+﻿using MediatR;
+using PatientChecking.Services.Appointment;
+using PatientChecking.Views.ViewModels;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace PatientChecking.Feature.Appointment.Queries
+{
+    public class GetAppointmentPagingQuery : IRequest<AppointmentListViewModel>
+    {
+        public int option { get; set; }
+        public int pageSize { get; set; }
+        public int pageIndex { get; set; }
+        class GetPatientByIdHandler : IRequestHandler<GetAppointmentPagingQuery, AppointmentListViewModel>
```

Comment:
Can we move this handler and other same places into a new file?

Reply:
Yes, Handler can be moved into a new file. However I have read the documentation about CQRS. Handler is declared with the query class so it's easy to see which handler will handle this query class.

Reply:
It can be separated files for finding easily. The 3rd party will scan the assembly to look up the handlers, it should be fine. You can put query and handler into a same file, but they should be separated classes, not in a same class